util/shared_lib.py

```diff
@@ -1,4 +1,5 @@
-from gpu import get_available_gpus
+from __future__ import print_function
+from .gpu import get_available_gpus
```

Comment:
It shouldn't need the absolute import *if* you don't change it to util.gpu. The reason you use .gpu is to make it an explicitly relative import, so you're not relying on absolute_import behaviour. The value of explicit relative import is that if you wind up moving the package (e.g. because "util" is not exactly an appropriate top-level package name if you're going to be installed in a system python install) the import doesn't need to change. Anyway, changed to absolute import.